sim/target-agnostic.mk

```diff
@@ -200,15 +214,36 @@ replace-rtl: $(fpga_v) $(ila_work_dir) $(fpga_vh) $(fpga_tcl_env)
 
 .PHONY: replace-rtl
 
-$(firesim_base_dir)/scripts/checkpoints/$(target_sim_tuple): $(fpga_work_dir)/stamp
+$(firesim_base_dir)/scripts/checkpoints/$(name_tuple): $(fpga_work_dir)/stamp
 	mkdir -p $(@D)
 	ln -sf $(fpga_build_dir)/checkpoints/to_aws $@
 
 # Runs a local fpga-bitstream build. Strongly consider using the manager instead.
 fpga: export CL_DIR := $(fpga_work_dir)
-fpga: $(fpga_v) $(base_dir)/scripts/checkpoints/$(target_sim_tuple)
+fpga: $(fpga_v) $(base_dir)/scripts/checkpoints/$(name_tuple)
 	cd $(fpga_build_dir)/scripts && ./aws_build_dcp_from_cl.sh -notify
 
+# Generate a script that can be run remotely on a build host by manager assuming only the following
+# are copied (recursively with Fabric's rsync_project) to the build-host:
+# - $(FAT_JAR)
+# - $(GENERATED_DIR)
+# - $(board_dir)/cl_firesim
+# The script will be run with 'bash -xe' to approximate the behavior of make shell execution
+#  --no-print-directory avoids some info print statements that aren't actual commands
+#  --assume-new=$(FAT_JAR) ensures that the script consistently generates the commands for
+#  $(FAT_JAR) -> replace-rtl, regardless of whether they need to be done.  This is done so that
+#  replace-rtl.sh should always be legit commands and we don't have to handle shortcuts for output
+#  that look like "make: Nothing to be done for `replace-rtl'."
+#  --assume-old=$(fpga_work_dir)/stamp ensures that the replace-rtl.sh will be generated including the
+#  recipe for $(fpga_work_dir)/stamp.  This is needed because  distributed-elaboration mode of
+#  'firesim buildafi' assumes that it will rsync $(board_dir)/cl_firesim to build host and
+#  replace-rtl.sh is responsible for creating $(fpga_work_dir) from the template.
+.PHONY: gen-replace-rtl-script
+gen-replace-rtl-script: $(PRE_ELABORATION_TARGETS) | $(GENERATED_DIR)
+	$(MAKE) replace-rtl -n  --no-print-directory --assume-old=$(fpga_work_dir)/stamp \
```

Comment:
ugh.  I pushed this too quickly without testing (or reading carefully enough about what --assume-old does).  --assume-old is not what we want here.  That tells make to assume that you can't rebuild that file because it is lost-knowledge from ancient times that you don't know how to remake.  I thought it was the opposite of --assume-new, i.e. assume that the file is older than it's prereqs and must always be rebuilt.

I have a fix that adds the first prereq of $(fpga_work_dir)/stamp into the --assume-new list and that forces make to always include the recipe for $(fpga_work_dir)/stamp in the replace-rtl.sh script.